lib/krb5/get_cred.c

```diff
@@ -963,6 +963,7 @@ get_cred_kdc_referral(krb5_context context,
     krb5_creds **referral_tgts = NULL;  /* used for loop detection */
     int loop = 0;
     int ok_as_delegate = 1;
+    int want_tgt = krb5_principal_is_krbtgt(context, in_creds->server);
```

Comment:
The reason for this is that we want to be able to use `kgetcred(1)` to get a `krbtgt/B@` using referrals.

It's only when we want `krbtgt/B@A` that we have to start by asking for a `krbtgt/A`.